yarn-project/telemetry-client/src/event_loop_monitor.ts

```diff
@@ -1,49 +1,119 @@
-import { promiseWithResolvers } from '@aztec/foundation/promise';
-import { Timer } from '@aztec/foundation/timer';
+import { type EventLoopUtilization, type IntervalHistogram, monitorEventLoopDelay, performance } from 'node:perf_hooks';
 
-import { EVENT_LOOP_LAG } from './metrics.js';
-import { type Meter, type ObservableGauge, type ObservableResult, ValueType } from './telemetry.js';
+import { NODEJS_EVENT_LOOP_STATE } from './attributes.js';
+import * as Metrics from './metrics.js';
+import {
+  type BatchObservableResult,
+  type Meter,
+  type ObservableGauge,
+  type UpDownCounter,
+  ValueType,
+} from './telemetry.js';
 
 /**
  * Detector for custom Aztec attributes
  */
 export class EventLoopMonitor {
-  private eventLoopLag: ObservableGauge;
+  private eventLoopDelayGauges: {
+    min: ObservableGauge;
+    max: ObservableGauge;
+    mean: ObservableGauge;
+    stddev: ObservableGauge;
+    p50: ObservableGauge;
+    p90: ObservableGauge;
+    p99: ObservableGauge;
+  };
+
+  private eventLoopUilization: ObservableGauge;
+  private eventLoopTime: UpDownCounter;
+
   private started = false;
 
-  constructor(meter: Meter) {
-    this.eventLoopLag = meter.createObservableGauge(EVENT_LOOP_LAG, {
-      unit: 'us',
-      valueType: ValueType.INT,
+  private lastELU: EventLoopUtilization | undefined;
+  private eventLoopDelay: IntervalHistogram;
+
+  constructor(private meter: Meter) {
+    const nsObsGauge = (name: (typeof Metrics)[keyof typeof Metrics], description: string) =>
+      meter.createObservableGauge(name, {
+        unit: 'ns',
+        valueType: ValueType.INT,
+        description,
+      });
+
+    this.eventLoopDelayGauges = {
+      min: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_MIN, 'Minimum delay of the event loop'),
+      mean: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_MEAN, 'Mean delay of the event loop'),
+      max: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_MAX, 'Max delay of the event loop'),
+      stddev: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_STDDEV, 'Stddev delay of the event loop'),
+      p50: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_P50, 'P50 delay of the event loop'),
+      p90: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_P90, 'P90 delay of the event loop'),
+      p99: nsObsGauge(Metrics.NODEJS_EVENT_LOOP_DELAY_P99, 'P99 delay of the event loop'),
+    };
```

Comment:
Curious: is it better to have multiple gauges, or a single gauge where we differentiate measurements using an attribute?

Reply:
I think it depends on what we're measuring, in this case I feel like min/max/mean/percentiles measure different things even though the source of data is the same